arelle/plugin/validate/NL/PluginValidationDataExtension.py

```diff
@@ -136,6 +144,21 @@ def checkContexts(self, modelXbrl: ModelXbrl) -> ContextData:
             contextsWithSegments=contextsWithSegments,
         )
 
+    def checkFilingDTS(
+            self,
+            val: ValidateXbrl,
+            modelDocument: ModelDocument,
+            visited: list[ModelDocument]
+    ) -> None:
+        visited.append(modelDocument)
+        for referencedDocument, modelDocumentReference in modelDocument.referencesDocument.items():
+            if referencedDocument not in visited and referencedDocument.inDTS:
+                self.checkFilingDTS(val, referencedDocument, visited)
+        if modelDocument.type == ModelDocumentFile.Type.SCHEMA:
+            for doc, docRef in modelDocument.referencesDocument.items():
+                if "import" in docRef.referenceTypes:
+                    val.extensionImportedUrls.add(doc.uri)
```

Comment:
I will probably end up pulling this into the data structure I'm building for 4.1.1.1 and 4.1.1.2, but this will make that easy.